llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -17571,6 +17571,11 @@ SDValue DAGCombiner::visitFCOPYSIGN(SDNode *N) {
                            APInt::getSignMask(SignVT.getScalarSizeInBits())))
     return SDValue(N, 0);
 
+  // We only take the non-sign bits from the value operand
+  if (SimplifyDemandedBits(N0,
```

Comment:
This looks like it could subsume all these optimizations from above:
```
  // copysign(fabs(x), y) -> copysign(x, y)
  // copysign(fneg(x), y) -> copysign(x, y)
  // copysign(copysign(x,z), y) -> copysign(x, y)
```

Reply:
Probably, but SimplifyDemandedBits doesn't handle fabs, fneg or copysign now